### DISCUSSION TITLE
Thoughts on treating use of deprecated things as problems?

### DISCUSSION THREAD

**Discussion by starball5** [microsoft/pylance-release#4713] [Q&A] [answered]
This just got asked about on Stack Overflow: https://stackoverflow.com/q/76883362/11107541. If I understand correctly, the use-case is to be able to more easily see all the usages of deprecated things in the workspace folder.

There's a similar issue ticket specific to TypeScript here: https://github.com/microsoft/vscode/issues/141209.

I also see something with similar keywords here: https://github.com/microsoft/vscode/issues/78226, but I'm not sure how related this is. It seems to be related to a "diagnostics" feature of VS Code that I'm not familiar with.

I saw https://github.com/microsoft/pylance-release/discussions/2714, but I don't think that's really related to this question.

**Comment by erictraut** [comment, marked as answer]
Pylance already has the capability. You can enable the `reportDeprecated` diagnostic check and specify whether you want these to appear as errors, warnings or infos. For a full list of available diagnostic checks, refer to [this documentation](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-rule-defaults).

**Reply by JavierSanchezCastro** [reply]
Thanks! It works as expected!!

**Comment by stuoningur** [comment]
I have the same issue, but when adding pyrightconfig.json many settings are disabled, as described here: https://github.com/microsoft/pylance-release/wiki/Settings.json-overridden-by-Pyrightconfig.json-or-Pyproject.toml

What is the proper solution to report deprecated as warnings/errors?

**Reply by rchiodo** [reply]
"reportDeprecated" should work in a pyrightconfig.json too.

Like so:

```json
    "reportDeprecated": "warning"
```

**Reply by stuoningur** [reply]
Yes, but then as I mentioned above, other settings are being disabled/overwritten, as described in the wiki link.

**Reply by rchiodo** [reply]
They're disabled if you put them in the settings.json. Not if they're in the pyrightconfig.json. It takes precedence.

**Reply by rchiodo** [reply]
This section here is saying that same thing:
https://github.com/microsoft/pylance-release/wiki/Settings.json-overridden-by-Pyrightconfig.json-or-Pyproject.toml#what-can-be-done-about-it

**Reply by stuoningur** [reply]
I tried different things, but just can't get it to work. As soon as I add the pyrightconfig.json, VS Code stops showing errors/warnings for the affected settings, no matter how I try to add them to the .json config
